prov/net/src/xnet_progress.c

```diff
@@ -45,6 +45,57 @@
 
 static ssize_t (*xnet_start_op[ofi_op_write + 1])(struct xnet_ep *ep);
 
+#ifdef HAVE_LIBURING
```

Comment:
I thought you wanted the `net` provider to manage the `io_uring` directly (no abstraction). 
I think it makes sense to initialize the `io_uring in the `net` provider as it may request specific flags during initialization. I'm not sure we should defined abstractions for that in the common code.

Reply:
I was comparing this to the ofi_epoll calls.  We make those directly from the providers, but the abstraction hides the implementation, or if it's available.  I thought I saw that you had abstractions -- something like ofi_uring_init() -- that the provider could just call.

Reply:
Ah, I think I initially misunderstood your statement:
> We obviously need an OS abstraction for an io_uring. I would actually introduce a typedef for this, similar to epoll. E.g. struct io_uring ofi_io_uring_t. This abstraction would be mostly a wrapper, with more control given to the provider
By OS abstraction you mean an abstraction in `src/iouring.c`. My bad, will fix it.